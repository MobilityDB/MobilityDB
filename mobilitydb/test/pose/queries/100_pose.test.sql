-------------------------------------------------------------------------------
--
-- This MobilityDB code is provided under The PostgreSQL License.
-- Copyright (c) 2016-2025, Université libre de Bruxelles and MobilityDB
-- contributors
--
-- MobilityDB includes portions of PostGIS version 3 source code released
-- under the GNU General Public License (GPLv2 or later).
-- Copyright (c) 2001-2025, PostGIS contributors
--
-- Permission to use, copy, modify, and distribute this software and its
-- documentation for any purpose, without fee, and without a written
-- agreement is hereby granted, provided that the above copyright notice and
-- this paragraph and the following two paragraphs appear in all copies.
--
-- IN NO EVENT SHALL UNIVERSITE LIBRE DE BRUXELLES BE LIABLE TO ANY PARTY FOR
-- DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING
-- LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION,
-- EVEN IF UNIVERSITE LIBRE DE BRUXELLES HAS BEEN ADVISED OF THE POSSIBILITY
-- OF SUCH DAMAGE.
--
-- UNIVERSITE LIBRE DE BRUXELLES SPECIFICALLY DISCLAIMS ANY WARRANTIES,
-- INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
-- AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS ON
-- AN "AS IS" BASIS, AND UNIVERSITE LIBRE DE BRUXELLES HAS NO OBLIGATIONS TO
-- PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
--
-------------------------------------------------------------------------------

-------------------------------------------------------------------------------
-- Input
-------------------------------------------------------------------------------

SELECT asText(pose 'Pose(Point(1 1),0.5)');
SELECT asText(pose ' pose   (  Point  ( 1  1  ) ,	0.5   )   ');
/* Errors */
SELECT pose 'point(1,0.5)';
SELECT pose 'pose 1,0.5)';
SELECT pose 'Pose(Point(1 1),0.5';
SELECT pose 'Pose(Point(1 1) 0.5)';
SELECT pose 'Pose(Point(1 1)000,0.5)';
SELECT pose 'Pose(Point(1 1),-1.5)';
SELECT pose 'Pose(Point(1 1),0.5)xxx';

-------------------------------------------------------------------------------
-- Constructors
-------------------------------------------------------------------------------

SELECT asText(pose('Point(1 1)', 0.5));
SELECT asText(pose(ST_Point(1,1), 0.5));
/* Errors */
SELECT pose('Linestring(1 1,2 2)',1.5);
SELECT pose('Point Z(1 1 1)',1.5);
SELECT pose('Point M(1 1 1)',1.5);
SELECT pose(geography 'Point(1 1)',1.5);
SELECT pose('Point(1 1)',-1.5);

-------------------------------------------------------------------------------
-- Accessing values
-------------------------------------------------------------------------------

SELECT ST_AsText(point(pose 'Pose(Point(1 1),0.5)'));
SELECT rotation(pose 'Pose(Point(1 1),0.5)');
SELECT srid(pose 'Pose(SRID=5676;Point(1 1),0.5)');

-------------------------------------------------------------------------------
-- Modification functions
-------------------------------------------------------------------------------

SELECT asText(round(pose 'Pose(Point(1.123456789 1.123456789), 0.123456789)', 6));

-------------------------------------------------------------------------------
-- Cast functions 
-------------------------------------------------------------------------------

SELECT ST_AsText(round(pose 'Pose(Point(1 1),0.2)'::geometry, 6));

-- SELECT geometry 'SRID=5676;Point(610.455019399524 528.508247341961)'::pose;

-- NULL

-- /* Errors */

-------------------------------------------------------------------------------
-- Comparisons
-------------------------------------------------------------------------------

-- true
SELECT pose 'Pose(Point(1.000001 1),0.5)' ~= pose 'Pose(Point(1 1),0.5)';
SELECT pose 'Pose(Point(1 1.000001),0.5)' ~= pose 'Pose(Point(1 1),0.5)';
SELECT pose 'Pose(Point(1 1),0.5000001)' ~= pose 'Pose(Point(1 1),0.5)';
-- false
SELECT pose 'Pose(Point(1.00001 1),0.5)' ~= pose 'Pose(Point(1 1),0.5)';
SELECT pose 'Pose(Point(1 1.00001),0.5)' ~= pose 'Pose(Point(1 1),0.5)';
SELECT pose 'Pose(Point(1 1),0.500001)' ~= pose 'Pose(Point(1 1),0.5)';

-------------------------------------------------------------------------------

SELECT pose 'Pose(Point(1 1),0.5)' = pose 'Pose(Point(1 1),0.5)';
SELECT pose 'Pose(Point(1 1),0.5)' = pose 'Pose(Point(1 1),0.7)';
SELECT pose 'Pose(Point(1 1),0.5)' = pose 'Pose(Point(2 2),0.5)';

SELECT pose 'Pose(Point(1 1),0.5)' != pose 'Pose(Point(1 1),0.5)';
SELECT pose 'Pose(Point(1 1),0.5)' != pose 'Pose(Point(1 1),0.7)';
SELECT pose 'Pose(Point(1 1),0.5)' != pose 'Pose(Point(2 2),0.5)';

SELECT pose 'Pose(Point(1 1),0.5)' < pose 'Pose(Point(1 1),0.5)';
SELECT pose 'Pose(Point(1 1),0.5)' < pose 'Pose(Point(1 1),0.7)';
SELECT pose 'Pose(Point(1 1),0.5)' < pose 'Pose(Point(2 2),0.5)';

SELECT pose 'Pose(Point(1 1),0.5)' <= pose 'Pose(Point(1 1),0.5)';
SELECT pose 'Pose(Point(1 1),0.5)' <= pose 'Pose(Point(1 1),0.7)';
SELECT pose 'Pose(Point(1 1),0.5)' <= pose 'Pose(Point(2 2),0.5)';

SELECT pose 'Pose(Point(1 1),0.5)' > pose 'Pose(Point(1 1),0.5)';
SELECT pose 'Pose(Point(1 1),0.5)' > pose 'Pose(Point(1 1),0.7)';
SELECT pose 'Pose(Point(1 1),0.5)' > pose 'Pose(Point(2 2),0.5)';

SELECT pose 'Pose(Point(1 1),0.5)' >= pose 'Pose(Point(1 1),0.5)';
SELECT pose 'Pose(Point(1 1),0.5)' >= pose 'Pose(Point(1 1),0.7)';
SELECT pose 'Pose(Point(1 1),0.5)' >= pose 'Pose(Point(2 2),0.5)';

-------------------------------------------------------------------------------/
