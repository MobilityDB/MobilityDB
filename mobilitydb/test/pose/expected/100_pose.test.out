SELECT asText(pose 'Pose(Point(1 1),0.5)');
        astext        
----------------------
 Pose(POINT(1 1),0.5)
(1 row)

SELECT asText(pose ' pose   (  Point  ( 1  1  ) ,	0.5   )   ');
        astext        
----------------------
 Pose(POINT(1 1),0.5)
(1 row)

/* Errors */
SELECT pose 'point(1,0.5)';
ERROR:  Could not parse cbuffer value: Missing prefix 'Pose'
LINE 2: SELECT pose 'point(1,0.5)';
                    ^
SELECT pose 'pose 1,0.5)';
ERROR:  Could not parse pose value: Missing opening parenthesis
LINE 1: SELECT pose 'pose 1,0.5)';
                    ^
SELECT pose 'Pose(Point(1 1),0.5';
ERROR:  Could not parse pose value: Missing closing parenthesis
LINE 1: SELECT pose 'Pose(Point(1 1),0.5';
                    ^
SELECT pose 'Pose(Point(1 1) 0.5)';
ERROR:  Missing delimeter character ',': Point(1 1) 0.5)
LINE 1: SELECT pose 'Pose(Point(1 1) 0.5)';
                    ^
SELECT pose 'Pose(Point(1 1)000,0.5)';
ERROR:  parse error - invalid geometry
LINE 1: SELECT pose 'Pose(Point(1 1)000,0.5)';
                    ^
HINT:  "Point(1 1)000" <-- parse error at position 14 within geometry
SELECT pose 'Pose(Point(1 1),-1.5)';
                          pose                           
---------------------------------------------------------
 POSE (0101000000000000000000F03F000000000000F03F, -1.5)
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)xxx';
ERROR:  Could not parse pose value: Extraneous characters at the end
LINE 1: SELECT pose 'Pose(Point(1 1),0.5)xxx';
                    ^
SELECT asText(pose('Point(1 1)', 0.5));
        astext        
----------------------
 Pose(POINT(1 1),0.5)
(1 row)

SELECT asText(pose(ST_Point(1,1), 0.5));
        astext        
----------------------
 Pose(POINT(1 1),0.5)
(1 row)

/* Errors */
SELECT pose('Linestring(1 1,2 2)',1.5);
                          pose                          
--------------------------------------------------------
 POSE (0101000000000000000000F03F000000000000F03F, 1.5)
(1 row)

SELECT pose('Point Z(1 1 1)',1.5);
ERROR:  The geometry cannot have Z dimension
SELECT pose('Point M(1 1 1)',1.5);
ERROR:  The geometry cannot have M dimension
SELECT pose(geography 'Point(1 1)',1.5);
ERROR:  function pose(geography, numeric) does not exist
LINE 1: SELECT pose(geography 'Point(1 1)',1.5);
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
SELECT pose('Point(1 1)',-1.5);
                          pose                           
---------------------------------------------------------
 POSE (0101000000000000000000F03F000000000000F03F, -1.5)
(1 row)

SELECT ST_AsText(point(pose 'Pose(Point(1 1),0.5)'));
 st_astext  
------------
 POINT(1 1)
(1 row)

SELECT rotation(pose 'Pose(Point(1 1),0.5)');
 rotation 
----------
      0.5
(1 row)

SELECT srid(pose 'Pose(SRID=5676;Point(1 1),0.5)');
 srid 
------
 5676
(1 row)

SELECT asText(round(pose 'Pose(Point(1.123456789 1.123456789), 0.123456789)', 6));
                 astext                  
-----------------------------------------
 Pose(POINT(1.123457 1.123457),0.123457)
(1 row)

SELECT ST_AsText(round(pose 'Pose(Point(1 1),0.2)'::geometry, 6));
 st_astext  
------------
 POINT(1 1)
(1 row)

SELECT pose 'Pose(Point(1.000001 1),0.5)' ~= pose 'Pose(Point(1 1),0.5)';
 ?column? 
----------
 t
(1 row)

SELECT pose 'Pose(Point(1 1.000001),0.5)' ~= pose 'Pose(Point(1 1),0.5)';
 ?column? 
----------
 t
(1 row)

SELECT pose 'Pose(Point(1 1),0.5000001)' ~= pose 'Pose(Point(1 1),0.5)';
 ?column? 
----------
 t
(1 row)

SELECT pose 'Pose(Point(1.00001 1),0.5)' ~= pose 'Pose(Point(1 1),0.5)';
 ?column? 
----------
 f
(1 row)

SELECT pose 'Pose(Point(1 1.00001),0.5)' ~= pose 'Pose(Point(1 1),0.5)';
 ?column? 
----------
 f
(1 row)

SELECT pose 'Pose(Point(1 1),0.500001)' ~= pose 'Pose(Point(1 1),0.5)';
 ?column? 
----------
 f
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' = pose 'Pose(Point(1 1),0.5)';
 ?column? 
----------
 t
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' = pose 'Pose(Point(1 1),0.7)';
 ?column? 
----------
 f
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' = pose 'Pose(Point(2 2),0.5)';
 ?column? 
----------
 f
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' != pose 'Pose(Point(1 1),0.5)';
 ?column? 
----------
 f
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' != pose 'Pose(Point(1 1),0.7)';
 ?column? 
----------
 t
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' != pose 'Pose(Point(2 2),0.5)';
 ?column? 
----------
 t
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' < pose 'Pose(Point(1 1),0.5)';
 ?column? 
----------
 f
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' < pose 'Pose(Point(1 1),0.7)';
 ?column? 
----------
 t
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' < pose 'Pose(Point(2 2),0.5)';
 ?column? 
----------
 f
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' <= pose 'Pose(Point(1 1),0.5)';
 ?column? 
----------
 t
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' <= pose 'Pose(Point(1 1),0.7)';
 ?column? 
----------
 t
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' <= pose 'Pose(Point(2 2),0.5)';
 ?column? 
----------
 f
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' > pose 'Pose(Point(1 1),0.5)';
 ?column? 
----------
 f
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' > pose 'Pose(Point(1 1),0.7)';
 ?column? 
----------
 f
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' > pose 'Pose(Point(2 2),0.5)';
 ?column? 
----------
 t
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' >= pose 'Pose(Point(1 1),0.5)';
 ?column? 
----------
 t
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' >= pose 'Pose(Point(1 1),0.7)';
 ?column? 
----------
 f
(1 row)

SELECT pose 'Pose(Point(1 1),0.5)' >= pose 'Pose(Point(2 2),0.5)';
 ?column? 
----------
 t
(1 row)

